### PR TITLE
fix: Clear error message when provided config is not a text file

### PR DESCRIPTION
**Pull request by sspaink**
resolve: https://github.com/influxdata/telegraf/issues/8753

The issue was providing a non text file to Telegraf as a config would cause the error ` Error parsing data: line 1: invalid TOML syntax` which isn't that helpful. I couldn't find a good way to detect the type of a file without reading it, you could detect it by extension: https://pkg.go.dev/mime?utm_source=godoc#TypeByExtension but testing this on windows I wouldn't get a result for `.toml`. So I chose to use: https://pkg.go.dev/golang.org/x/tools/godoc/util?utm_source=godoc#IsText which will look at the contents of the file to determine if it is a text file.

This does mean that depending on the binary size it could be slow to print this error message, but hopefully the error message will help regardless. Because when providing a URL the request header is updated with `req.Header.Add("Accept", "application/toml")`, I thought this might suffice and the extra check if it is a text file shouldn't be necessary and its the servers responsibility. But open to consider this differently.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -26,6 +29,22 @@ import (
...
+	cmd := exec.Command("go", "build", "-o", binaryFile, "./tools/custom_builder")
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+	err := cmd.Run()
+	require.NoError(t, err, fmt.Sprintf("stdout: %s, stderr: %s", outb.String(), errb.String()))
+	c := NewConfig()
+	err = c.LoadConfig(binaryFile)
+	require.ErrorContains(t, err, "provided config is not a text file")
+	require.Error(t, err)
```
Comment (Hipska) [inline config/config_test.go:45]: Shouldn't these two be swapped, I think otherwise, the last one doesn't make much sense?

**Review by Hipska — COMMENTED** [review]
> when providing a URL the request header is updated with req.Header.Add("Accept", "application/toml"), I thought this might suffice and the extra check if it is a text file shouldn't be necessary and its the servers responsibility.

I agree with this, they will still receive the message that it isn't valid TOML.

File: config/config.go [hunk 2]
```diff
@@ -581,7 +582,16 @@ func LoadConfigFile(config string) ([]byte, error) {
...
 
 	// If it isn't a https scheme, try it as a file
-	return os.ReadFile(config)
+	buffer, err := os.ReadFile(config)
+	if err != nil {
+		return nil, err
+	}
+
+	if !util.IsText(buffer) {
+		return nil, fmt.Errorf("provided config is not a text file")
```
Comment (powersj) [inline config/config.go:591]: Can we do something like this to be clear as to what file we are talking about?

```suggestion
		return nil, fmt.Errorf("provided config is not a TOML file: %s", config)
```

**Review by Hipska — APPROVED** [review]

**Comment by sspaink** [comment]
Weird, the tests in Windows are failing in the CI because of this change but I can't reproduce it locally on a Windows machine (https://app.circleci.com/pipelines/github/influxdata/telegraf/12585/workflows/9895e703-3964-4ddf-ae2a-91aef8ecaab1/jobs/203622). I thought maybe it had to do with the encoding because `util.IsTest` requires utf8 but before failing it prints the mimetype correctly as `text/plain; charset=utf-8`.

Updated the logic to use the mimetype returned by `http.DetectContentType` as a potential alternative.

**Review by powersj — APPROVED** [review]

**Pull request merged**